src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityCredentialService .cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Utils;
+using Microsoft.Identity.Client.OAuth2;
+using System.Net.Http;
+using System.Security.Cryptography.X509Certificates;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Identity.Client.ManagedIdentity
+{
+    internal class ManagedIdentityCredentialService
+    {
+        private readonly ManagedIdentityRequest _request;
+        private readonly X509Certificate2 _bindingCertificate;
+        private readonly RequestContext _requestContext;
```

Comment:
Try to avoid passing in "RequesetContext" around. It's starting to become a https://en.wikipedia.org/wiki/God_object

A better pattern is to pass in only what is needed - the logger and the httpManager in this case.